lib/takeaway.rb

```diff
@@ -0,0 +1,79 @@
+require "csv"
+require_relative 'dish_list'
+require_relative 'dish'
+require_relative 'order'
+require_relative 'sms'
+
+class Takeaway
+
+  attr_reader :dish_list, :order
+
+  def initialize( args = {dish_list: DishList.new(Dish), order: Order.new} )
+    @dish_list     = args[:dish_list]
+    @order         = args[:order]
+    @placed_orders = []
+  end
+
+  def load_dishes(filename='menu.csv')
+    if File.exist?(filename)
+      CSV.foreach(filename) { |data| dish_list.add_dish(data) }
+    else
+      fail "File '#{filename}' not found!"
+    end
+  end
+
+  def menu
+    puts dish_list
+  end
+
+  def add_dish(id, qty=1)
+    fail_if_wrong_id(id)
+    fail_if_wrong_qty(qty)
+    dish = dish_list.select_dish(id)
+    order.add_item(dish, qty)
+  end
+
+  def remove_dish(id, qty=1)
+    fail_if_wrong_id(id)
+    fail_if_wrong_qty(qty)
+    dish = dish_list.select_dish(id)
+    order.remove_item(dish, qty)
+  end
+
+  def order_total
+    order.total
+  end
+
+  def place_order
+    fail "You haven't added any items to your order" if order.items.empty?
+    @placed_orders << order
+    send_confirmation_sms
+    @order = Order.new
+    confirmation_text
+  end
+
+  def placed_orders
+    @placed_orders.dup
+  end
+
+  def confirmation_text
+    h = Time.now.hour
+    delivery_time = "#{h==23 ? 0 : h+1}:#{Time.now.min}"
```

Comment:
`Time` already provides this functionality.  `Time.now + 3600` will add 3600 seconds (i.e. 1 hour)